packages/wrangler/src/match-tag.ts

```diff
@@ -0,0 +1,46 @@
+import { fetchResult } from "./cfetch";
+import { getCIMatchTag } from "./environment-variables/misc-variables";
+import { FatalError } from "./errors";
+import { logger } from "./logger";
+import type { ServiceMetadataRes } from "./init";
+
+export async function verifyWorkerMatchesCITag(
+	accountId: string,
+	workerName: string
+) {
+	const matchTag = getCIMatchTag();
+
+	// If no tag is provided through the environment, nothing needs to be verified
+	if (!matchTag) {
+		return;
+	}
+
+	let tag;
+
+	try {
+		const worker = await fetchResult<ServiceMetadataRes>(
+			`/accounts/${accountId}/workers/services/${workerName}`
+		);
+		tag = worker.default_environment.script.tag;
+	} catch (e) {
+		logger.debug(e);
+		// code: 10090, message: workers.api.error.service_not_found
+		if ((e as { code?: number }).code === 10090) {
+			throw new FatalError(
+				`Your Worker's name (${workerName}) does not match what is expected by the CI system`
```

Comment:
is there an action we can recommend to users here?

Reply:
We do have some error messaging written up [here](https://jira.cfdata.org/browse/BANDA-411). cc @aninibread